src/pages/AddTransactionForm.jsx

```diff
@@ -61,12 +61,18 @@ export default function AddTransactionForm({fetchTransactions}) {
 
     const handleSubmit = () => {
       setOpen(false);
-      if(newAmount){
+      if(newAmount && newDescription.length < 50 && newDescription.length > 0 ){
         const absAmount = Math.abs(parseFloat(newAmount).toFixed(2))
         addTransactions(newDescription,absAmount,transactionDate)
+      }else if(newAmount && (newDescription.length > 50 || newDescription.length === 0)){
+        alert("fail to add new transaction, the description is too long or empty")
```

Comment:
Update alert text to "Failed to add a new transaction, the description you entered is too long or empty"

Or you can separate the length conditional and empty conditional to give more precise alerts. Up to you!

For example (pseudocode):
`if(new.Description.length==0) { alert("Failed to add a new transaction, the description you entered is empty")} else if(new.Description.length < 50 ) { alert("Failed to add a new transaction, the description you entered is too long")}`